service/src/main/java/bio/terra/cbas/dependencies/sam/SamService.java

```diff
@@ -54,4 +103,16 @@ public Result checkHealth() {
       return new Result(false, e.getMessage());
     }
   }
+
+  public boolean hasReadPermission() {
```

Comment:
Opening up to discussion - what do we think about `checkReadPermission` instead of `hasReadPermission`, where `checkReadPermission` throws an exception if user doesn't have permission and does nothing if user has permission. Then we wouldn't need to explicitly throw the same exception at the top of each endpoint.

Reply:
The SamException would still be displayed differently from the ForbiddenException even if we throw ForbiddenException from inside SamService. I do agree that this method allows for better testing and wider use cases, I'm just not sure how many cases we will want to check "does the user have the given access" and proceed even if the user doesn't. Also interested in hearing from other reviewers

Reply:
@kpierre13 as the second reviewer can you share your thoughts on this?

Reply:
That was my thought. `hasXPermission` is beneficial if we want to check access but not throw, but I think we will always want to throw?

Reply:
The only place I can think of at the moment where we don't want to simply throw an exception is when we want to update the Run and Run set status after getting a callback from Cromwell via the new callback API. In that case, after we check if the user still has write permission, we still might want to clean up the status of Run/Workflow to Failed and possibly return a new/combined error back to user saying that they didn't have permission to update status of Run X. 🤔

Reply:
Ah, that makes sense. In that case I'm good with leaving as-is